### PR TITLE
Standardize interface naming for indexer types

### PR DESCRIPTION
**Pull request by frankfka**

### CODE REVIEW

File: packages/indexer-client/src/IndexerBaseClient.ts [hunk 1]
```diff
@@ -103,8 +103,8 @@ export class IndexerBaseClient {
    * @param params
    */
   async listSubaccounts(
```
Comment (yruej301) [inline packages/indexer-client/src/IndexerBaseClient.ts:105]: minor but how about we just call this `getSubaccounts` and rename params/response to just `IndexerQuerySubaccountsParams` / `IndexerQuerySubaccountsResponse`. seems more standard.

File: packages/indexer-client/src/IndexerClient.ts [hunk 2]
```diff
@@ -7,36 +7,36 @@ import { toBigDecimal } from '@vertex-protocol/utils';
...
+  IndexerQuerySubaccountSettlementEventsResponse,
   IndexerSettlementEvent,
   PaginatedIndexerEventsResponse,
 } from './types';
 
 export class IndexerClient extends IndexerBaseClient {
   async getPaginatedSubaccountMatchEvents(
-    params: GetIndexerSubaccountMatchEventParams,
-  ): Promise<GetIndexerSubaccountMatchEventsResponse> {
+    params: IndexerQuerySubaccountMatchEventParams,
```
Comment (yruej301) [inline packages/indexer-client/src/IndexerClient.ts:38]: it seems some params / response include `paginated` e.g: `IndexerQueryPaginatedOrdersParams` and some don't, think we should standardize it.

File: packages/indexer-client/src/IndexerClient.ts [hunk 3]
```diff
@@ -183,8 +183,8 @@ export class IndexerClient extends IndexerBaseClient {
   }
 
   async getPaginatedSubaccountOrders(
-    params: GetIndexerPaginatedOrdersParams,
-  ): Promise<GetIndexerPaginatedOrdersResponse> {
+    params: IndexerQueryPaginatedOrdersParams,
```
Comment (yruej301) [inline packages/indexer-client/src/IndexerClient.ts:186]:
```suggestion
    params: IndexerQuerySubaccountOrdersParams,
```

File: packages/indexer-client/src/IndexerClient.ts [hunk 4]
```diff
@@ -183,8 +183,8 @@ export class IndexerClient extends IndexerBaseClient {
...
+  ): Promise<IndexerQueryPaginatedOrdersResponse> {
```
Comment (yruej301) [inline packages/indexer-client/src/IndexerClient.ts:187]:
```suggestion
  ): Promise<IndexerQuerySubaccountOrdersResponse> {
```

**Review by yruej301 — COMMENTED** [review]
overall the renaming looks good, left some minor comments

File: packages/indexer-client/src/IndexerClient.ts (hunk 2 above)
Comment (frankfka) [reply packages/indexer-client/src/IndexerClient.ts:38]: Hm these are paginated queries with pagination `meta`, but the queries on `IndexerBaseClient` don't really have pagination built in

**Pull request closed without merging**